src/segment-loader.js

```diff
@@ -2277,7 +2277,8 @@ export default class SegmentLoader extends videojs.EventTarget {
         `video buffer: ${timeRangesToArray(videoBuffered).join(', ')}, `);
       this.error({
         message: 'Quota exceeded error with append of a single segment of content',
-        excludeUntil: Infinity
+        excludeUntil: Infinity,
+        errorType: videojs.Errors.SegmentExceedsBufferQuota
```

Comment:
`SegmentExceedsSourceBufferQuota`

Reply:
I noticed that we always use the following pattern:

      this.error(...);
      this.trigger('error');
      
      maybe we should move trigger to error method?

Reply:
I think the only issue for moving all of this into the error method would be that occasionally `error` is triggered, and other times `appendError` is triggered. We could create a separate `this.appendError` function that calls both `this.error(..)` and `this.trigger('appendError')` and clean this up a bit.